Changing module to use the port connection according to the Redis plan

l-lafin opened the pull request:
Currently, the object `cloudfoundry_service_key.key.credentials` properties will change according to the Redis plan, so I'm adding conditions according to the selected plan.

- [X] Dynamically using `port` or `sentinel_port` according to the Redis plan selected.
- [X] Updating readme to make explicit this module usage.
- [X] Updating usage example.

l-lafin commented:
Hi, @loafoe while I was using the latest version of the module I got an error stating that the object `credentials` does not contain the property 'sentinel_port' so I opened this PR to change to 'port', maybe is something related to the 'redis-development-standalone' plan.

loafoe commented:
Thanks for the PR! It looks like different plans return different credentials. I only tested it with cluster plans and not standalone. So your fix will work for standalone, but will break for cluster plans. The changes would need to support all plans. Can you look into this?

l-lafin commented:
Sure, I'll try to have a look at it this week :v:

l-lafin commented:
Hey @loafoe,
I changed to get the 'port' attribute according to the selected plan. Btw, I also updated the usage example because it was outdated and was still using the `metrics_port` and `metrics_app_id` (that doesn't exist anymore). Can you take a look if the example is still correct?

loafoe commented on main.tf:
```diff
@@ -1,5 +1,6 @@
 locals {
-  postfix = var.name_postfix != "" ? var.name_postfix : random_id.id.hex
+  postfix            = var.name_postfix != "" ? var.name_postfix : random_id.id.hex
+  planCredentialPort = replace(var.plan, "standalone", "") != var.plan ? "port" : "sentinel_port"
```
Here I would check the value of "port" instead of looking at the plan value

l-lafin replied:
Clarifying a bit, the trick is that the credential object is different according to the plan, so I'm getting errors likes "This object has no argument, nested block, or exported attribute named sentinel_port" when trying to get the value of the port, so even functions like try(), coalesce(), etc the error remains.

Working using the plan it's not ideal because we need to keep track of newer plans but it worked nice.
